### ISSUE TITLE
Return type error in TensorView::subview

### ISSUE THREAD

**Issue by qliu93** [NVIDIA/cutlass#173]
https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/tensor_view.h#L222

The return type of `TensorView::subview` is `TensorView`, but it calls `TensorView::add_coord_offset` whose return type is TensorRef.

An example:

```
int I = 3, J = 4, K = 5;
float* ptr = (float*)malloc(I * J * K * sizeof(float));

IdentityTensorLayout<3> layout(make_Coord(J * K, K, 1));
auto extent = make_Coord(I, J, K);
auto tensor_view = make_TensorView(ptr, layout, extent);

auto ret = tensor_view.subview(extent, make_Coord(1, 1, 1));
```

cutlass/include/cutlass/tensor_view.h(208): error: no suitable user-defined conversion from "cutlass::TensorRef<float, cutlass::IdentityTensorLayout<3>>" to "cutlass::TensorView<float, cutlass::IdentityTensorLayout<3>>" exists

**Comment by hwu36** [comment]
Thanks for catching this.

What about below?

```
    TensorView result(ref(), extent.clamp(extent_ - location));                                                                         
    result.add_pointer_offset(this->offset(b));
    return result;
```

**Comment by Peter9606** [comment]
> Thanks for catching this.
> 
> What about below?
> 
> ```
>     TensorView result(ref(), extent.clamp(extent_ - location));                                                                         
>     result.add_pointer_offset(this->offset(b));
>     return result;
> ```

There isn't a 'b' variable at all in the original code.

**Comment by hwu36** [comment]
> > Thanks for catching this.
> > What about below?
> > ```
> >     TensorView result(ref(), extent.clamp(extent_ - location));                                                                         
> >     result.add_pointer_offset(this->offset(b));
> >     return result;
> > ```
> 
> There isn't a 'b' variable at all in the original code.

You are right.  I fixed in the above reply.  Thank you.

**Comment by d-k-b** [comment]
Fixed in 2.5.
